packages/ember-toucan-form/src/components/toucan-form.ts

```diff
@@ -34,7 +31,7 @@ export default class ToucanFormComponent<
   DATA extends UserData,
   SUBMISSION_VALUE
 > extends Component<ToucanFormComponentSignature<DATA, SUBMISSION_VALUE>> {
-  TextareaFieldComponent = TextareaFieldComponent;
+  TextareaFieldComponent = TextareaFieldComponent<DATA>;
```

Comment:
Lost most time on this one. Same back then when working on headless-form, but forgot about this... 🙈😅

This is important to propagate the type of `DATA` down to the yielded components, otherwise TS/Glint would think that the `DATA` of the form could be of a different type than the `DATA` of e.g. the TextareaField...